pkg/backend/httpstate/diffs.go

```diff
@@ -113,7 +123,7 @@ func (dds *deploymentDiffState) Diff(ctx context.Context, deployment deployment)
 		checkpointHash = dds.computeHash(childCtx, after)
 	}()
 
-	delta, err := dds.computeEdits(childCtx, string(before), string(after))
+	delta, err := dds.computeEdits(childCtx, asString(before), asString(after))
```

Comment:
Side-note: I looked into what we're doing with this.
We're using a [(possibly) unmaintained library](https://github.com/hexops/gotextdiff) here, but we're also serializing its representation of the edit script into JSON and saving that. 
I'm not completely sure what we do with this edit script, but we should probably control that schema if it's crossing system boundaries.
I can create an issue about this if you can confirm that my guess here is correct.

(This was prompted by investigating whether the unsafe operation could just be moot if we added `[]byte` support to the library.)

Reply:
Yeah I agree with pretty much all of these points.

Even with the new library we'll still be using unsafe strings a bit so that we can use [`lcs.DiffSlices`](https://pkg.go.dev/github.com/pgavlin/diff@v0.0.0-20230507165749-bf75e9a892e2/lcs#DiffSlices). Given that time performance isn't necessarily a limiting factor here, we could instead use [`lcs.DiffAnySlices`](https://pkg.go.dev/github.com/pgavlin/diff@v0.0.0-20230507165749-bf75e9a892e2/lcs#DiffAnySlices) and hand it `json.RawMessage` slices and a comparator, though that implementation is somewhat slower due to some codegen issues I haven't tracked down.